lib/takeaway.rb

```diff
@@ -0,0 +1,13 @@
+require './lib/menu.rb'
+
+class Takeaway
+  attr_reader :menu
+  def initialize
+    @menu = Menu.new
```

Comment:
I'm impressed you definitely know what you're doing, especially with the corresponding tests

I'm no expert in dependency injection, but this article near the beginning

https://medium.com/@Bakku1505/introduction-to-dependency-injection-in-ruby-dc238655a278

explains how linking the classes together so "strongly" ie 'def initialize... @menu = Menu.new' can cause issues down the line